homeassistant/components/light/group.py

```diff
@@ -0,0 +1,275 @@
+        """Register callbacks."""
+        @callback
+        def async_state_changed_listener(entity_id: str, old_state: State,
+                                         new_state: State):
+            """Handle child updates."""
+            self.async_schedule_update_ha_state(True)
+
+        self._async_unsub_state_changed = async_track_state_change(
+            self.hass, self._entity_ids, async_state_changed_listener)
+
+    async def async_will_remove_from_hass(self):
+        """Callback when removed from HASS."""
+        if self._async_unsub_state_changed:
+            self._async_unsub_state_changed()
+            self._async_unsub_state_changed = None
+
+    @property
+    def name(self) -> str:
+        """Return the name of the entity."""
+        return self._name
+
+    @property
+    def state(self) -> str:
+        """Return the state."""
+        return self._state
+
+    @property
+    def is_on(self) -> bool:
+        """Return True if entity is on."""
+        return self._state == STATE_ON
+
+    @property
+    def brightness(self) -> Optional[int]:
+        """Return the brightness of this light between 0..255."""
+        return self._brightness
+
+    @property
+    def xy_color(self) -> Optional[Tuple[float, float]]:
+        """Return the XY color value [float, float]."""
+        return self._xy_color
+
+    @property
+    def rgb_color(self) -> Optional[Tuple[int, int, int]]:
+        """Return the RGB color value [int, int, int]."""
+        return self._rgb_color
+
+    @property
+    def color_temp(self) -> Optional[int]:
+        """Return the CT color value in mireds."""
+        return self._color_temp
+
+    @property
+    def min_mireds(self) -> Optional[int]:
+        """Return the coldest color_temp that this light supports."""
+        return self._min_mireds
+
+    @property
+    def max_mireds(self) -> Optional[int]:
+        """Return the warmest color_temp that this light supports."""
+        return self._max_mireds
+
+    @property
+    def white_value(self) -> Optional[int]:
+        """Return the white value of this light between 0..255."""
+        return self._white_value
+
+    @property
+    def effect_list(self) -> Optional[List[str]]:
+        """Return the list of supported effects."""
+        return self._effect_list
+
+    @property
+    def effect(self) -> Optional[str]:
+        """Return the current effect."""
+        return self._effect
+
+    @property
+    def supported_features(self) -> int:
+        """Flag supported features."""
+        return self._supported_features
+
+    @property
+    def should_poll(self) -> bool:
+        """No polling needed for a group light."""
+        return False
+
+    async def async_turn_on(self, **kwargs):
+        """Forward the turn_on command to all lights in the group."""
+        kwargs[ATTR_ENTITY_ID] = self._entity_ids
+        await light.async_turn_on(self.hass, blocking=True, **kwargs)
```

Comment:
*Theoretically* (and this is a *big* hypothetical) we could receive a `blocking` inside our own `kwargs`, then this would cause an exception `TypeError: function got multiple values for keyword argument blocking`, but I suppose it's better to just break if that happens.

Reply:
Like I stated above, call the service directly and make sure you know how the turn_on kwargs map to the service attributes.